### PR TITLE
site: add code-hide Button for CodePreviewer

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | 无需纳入ChangeLog |
| 🇨🇳 Chinese | 无需纳入ChangeLog |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at add1e37</samp>

Improved the code previewer functionality and readability for the dumi theme. Added a `hide code` button to `.dumi/theme/builtins/Previewer/CodePreviewer.tsx` using a hook and an icon. Refactored `.dumi/theme/common/CodePreview.tsx` to a new file, JSX syntax, and types.

### <samp>🤖 Generated by Copilot at add1e37</samp>

* Move and refactor the CodePreview component from `.dumi/theme/common/CodePreview.tsx` to `.dumi/theme/builtins/Previewer/CodePreviewer.tsx` ([link](https://github.com/ant-design/ant-design/pull/44449/files?diff=unified&w=0#diff-086f3227db976462d986107d6240e5fa7cba6d817b889ff580ea55b386950a9dL2-R5), [link](https://github.com/ant-design/ant-design/pull/44449/files?diff=unified&w=0#diff-086f3227db976462d986107d6240e5fa7cba6d817b889ff580ea55b386950a9dL20-R29))
* Add the React import statement to use React components and hooks in `CodePreviewer.tsx` ([link](https://github.com/ant-design/ant-design/pull/44449/files?diff=unified&w=0#diff-54c90701ed595205ea2cd4286b1cd0ee8b2f01f77a23088e0108feda7af2d65fR1))
* Import the VerticalAlignMiddleOutlined icon from @ant-design/icons and use it for a new button to hide the code preview section ([link](https://github.com/ant-design/ant-design/pull/44449/files?diff=unified&w=0#diff-54c90701ed595205ea2cd4286b1cd0ee8b2f01f77a23088e0108feda7af2d65fL6-R16), [link](https://github.com/ant-design/ant-design/pull/44449/files?diff=unified&w=0#diff-54c90701ed595205ea2cd4286b1cd0ee8b2f01f77a23088e0108feda7af2d65fL512-R529))
* Add custom styles and localization messages for the new button ([link](https://github.com/ant-design/ant-design/pull/44449/files?diff=unified&w=0#diff-54c90701ed595205ea2cd4286b1cd0ee8b2f01f77a23088e0108feda7af2d65fL512-R529))

### CODE REVIEW

**Comment by aliBenhenia** [comment]
can i see more details

**Comment by li-jia-nan** [comment]
1693127367444_20E7283B-2525-4c51-B0A7-83B2AD5B8E57

File: .dumi/theme/builtins/Previewer/CodePreviewer.tsx [hunk 1]
```diff
@@ -509,8 +511,21 @@ createRoot(document.getElementById('container')).render(<Demo />);
             sourceCode={entryCode}
             jsxCode={jsx}
             styleCode={style}
-            onCodeTypeChange={(type) => setCodeType(type)}
+            onCodeTypeChange={setCodeType}
           />
+          <Button
+            block
+            icon={<VerticalAlignMiddleOutlined style={{ fontSize: 16, marginRight: 6 }} />}
```
Comment (zombieJ) [inline .dumi/theme/builtins/Previewer/CodePreviewer.tsx:518]: 这里都要用 token~

File: .dumi/theme/common/styles/Demo.tsx [hunk 2]
```diff
@@ -217,6 +217,16 @@ const GlobalDemoStyles: React.FC = () => {
             &-expand {
               display: block;
             }
+            .code-hide-btn {
+              display: flex;
+              justify-content: center;
```
Comment (zombieJ) [inline .dumi/theme/common/styles/Demo.tsx:222]: block 的 button 不是本来就居中的么？

File: .dumi/theme/common/styles/Demo.tsx [hunk 3]
```diff
@@ -217,6 +217,16 @@ const GlobalDemoStyles: React.FC = () => {
...
+              align-items: center;
+              border-radius: 0 0 ${token.borderRadius}px ${token.borderRadius}px;
```
Comment (zombieJ) [inline .dumi/theme/common/styles/Demo.tsx:224]: 直接逻辑位置设置 0，其他的不改呢？

File: .dumi/theme/common/styles/Demo.tsx [hunk 4]
```diff
@@ -217,6 +217,16 @@ const GlobalDemoStyles: React.FC = () => {
...
+              .ant-btn-icon {
+                font-size: ${token.fontSizeLG}px;
```
Comment (zombieJ) [inline .dumi/theme/common/styles/Demo.tsx:226]: 魔改图标的目的是什么，原本的样式有问题么？

File: .dumi/theme/common/styles/Demo.tsx (hunk 2 above)
Comment (li-jia-nan) [reply .dumi/theme/common/styles/Demo.tsx:222]: 设置了 `font-size` 之后，垂直居中有点歪了，这里主要是加 `align-items`，写顺手了把 `justify-content` 也加上了

File: .dumi/theme/common/styles/Demo.tsx (hunk 4 above)
Comment (li-jia-nan) [reply .dumi/theme/common/styles/Demo.tsx:226]: 默认图标感觉有点小，并且距离文字有点近，所以调整了一下尺寸和间距

File: .dumi/theme/builtins/Previewer/CodePreviewer.tsx (hunk 1 above)
Comment (li-jia-nan) [reply .dumi/theme/builtins/Previewer/CodePreviewer.tsx]: done~

File: .dumi/theme/common/styles/Demo.tsx (hunk 3 above)
Comment (li-jia-nan) [reply .dumi/theme/common/styles/Demo.tsx]: done~

File: .dumi/theme/common/styles/Demo.tsx [hunk 5]
```diff
@@ -217,6 +217,17 @@ const GlobalDemoStyles: React.FC = () => {
...
             }
+            .code-hide-btn {
+              display: flex;
+              justify-content: center;
+              align-items: center;
+              border-start-start-radius: 0;
+              border-start-end-radius: 0;
+              .ant-btn-icon {
+                font-size: ${token.fontSizeLG}px;
+                margin-right: ${token.marginXS}px;
```
Comment (MadCcc) [inline .dumi/theme/common/styles/Demo.tsx:228]: 这不要写全局样式吧，用 createStyles 在 Previewr 里写，还能被拆包。

File: .dumi/theme/builtins/Previewer/CodePreviewer.tsx [hunk 6]
```diff
@@ -509,8 +511,16 @@ createRoot(document.getElementById('container')).render(<Demo />);
             sourceCode={entryCode}
             jsxCode={jsx}
             styleCode={style}
-            onCodeTypeChange={(type) => setCodeType(type)}
+            onCodeTypeChange={setCodeType}
           />
+          <Button
+            block
+            className="code-hide-btn"
+            icon={<VerticalAlignMiddleOutlined />}
```
Comment (MadCcc) [inline .dumi/theme/builtins/Previewer/CodePreviewer.tsx:519]: <VerticalAlignTopOutlined />
用这个吧

File: .dumi/theme/common/styles/Demo.tsx (hunk 5 above)
Comment (li-jia-nan) [reply .dumi/theme/common/styles/Demo.tsx]: done

File: .dumi/theme/builtins/Previewer/CodePreviewer.tsx (hunk 6 above)
Comment (li-jia-nan) [reply .dumi/theme/builtins/Previewer/CodePreviewer.tsx]: done

**Comment by MadCcc** [comment]
样式要再调一下，不要用 Button 吧，感觉不太适合

文案用“收起”即可，试一下 `<UpOutlined />` 或者 `<UpCircleOutlined />`，可能跟合适些

**Comment by Wxh16144** [comment]
image

隔壁 element-plus 的效果就很优秀，可以借鉴一下。 👇🏻👇🏻👇🏻

https://github.com/ant-design/ant-design/assets/32004925/e92f2ec3-4faa-4ce5-8296-85525c83fc4e

File: .dumi/theme/locales/en-US.json [hunk 7]
```diff
@@ -31,6 +31,7 @@
   "app.demo.copied": "Copied!",
   "app.demo.code.show": "Show code",
   "app.demo.code.hide": "Hide code",
+  "app.demo.code.hide.simplify": "Hide",
```
Comment (Wxh16144) [inline .dumi/theme/locales/en-US.json:34]: 这里的国际化岂不是重复了吗？

File: .dumi/theme/builtins/Previewer/CodePreviewer.tsx [hunk 8]
```diff
@@ -63,6 +66,33 @@ function useShowRiddleButton() {
...
+    codeHideBtn: css`
+      width: 100%;
+      height: 40px;
+      display: flex;
+      justify-content: center;
+      align-items: center;
+      border-start-start-radius: 0;
+      border-start-end-radius: 0;
+      border-end-start-radius: ${borderRadius}px;
+      border-end-end-radius: ${borderRadius}px;
```
Comment (MadCcc) [inline .dumi/theme/builtins/Previewer/CodePreviewer.tsx:81]:
```suggestion
      border-radiusL 0 0 ${borderRadius}px ${borderRadius}px;
```

Comment (MadCcc) [reply .dumi/theme/builtins/Previewer/CodePreviewer.tsx:81]: 再加个 `colorBgElevated` 背景色

File: .dumi/theme/builtins/Previewer/CodePreviewer.tsx [hunk 9]
```diff
@@ -509,8 +541,12 @@ createRoot(document.getElementById('container')).render(<Demo />);
             sourceCode={entryCode}
             jsxCode={jsx}
             styleCode={style}
-            onCodeTypeChange={(type) => setCodeType(type)}
+            onCodeTypeChange={setCodeType}
           />
+          <div className={styles.codeHideBtn} onClick={() => setCodeExpand(false)}>
```
Comment (MadCcc) [inline .dumi/theme/builtins/Previewer/CodePreviewer.tsx:546]: 可以加个 tabIndex

File: .dumi/theme/locales/en-US.json (hunk 7 above)
Comment (li-jia-nan) [reply .dumi/theme/locales/en-US.json:34]: 不这么做的话只能写一堆判断的逻辑了，这里统一处理比较好，用 simplify 后缀表示简化的文案

File: .dumi/theme/builtins/Previewer/CodePreviewer.tsx (hunk 8 above)
Comment (li-jia-nan) [reply .dumi/theme/builtins/Previewer/CodePreviewer.tsx]: done

File: .dumi/theme/builtins/Previewer/CodePreviewer.tsx (hunk 9 above)
Comment (li-jia-nan) [reply .dumi/theme/builtins/Previewer/CodePreviewer.tsx:543]: done

**Review by MadCcc — APPROVED** [review]

**Pull request merged**